components/datetime/src/options/components.rs

```diff
@@ -341,6 +342,7 @@ impl Bag {
     derive(Serialize, Deserialize),
     serde(rename_all = "kebab-case")
 )]
+#[allow(clippy::exhaustive_enums)] // this type is stable
```

Comment:
Nit: I think these should be `non_exhaustive` because they could change based on if CLDR or ECMA-402 move to add more styles. This has been done in the past (see: hour cycle, time zone name).